### PR TITLE
Fix high byte unicode character encoding

### LINKED ISSUE #196: JSON output not correctly escaping non-Unicode characters
The JSON RFC has this range of unescaped characters for strings:
`      unescaped = %x20-21 / %x23-5B / %x5D-10FFFF`
but also the text "All Unicode characters may be placed within the quotation marks".

In yajl there is this code: https://github.com/lloyd/yajl/blob/5e3a7856e643b4d6410ddc3f84bc2f38174f2872/src/yajl_encode.c#L61
which is correctly escaping below 0x20, but characters above 0x80 are not escaped. Characters above 0x80 however need to be escaped to be proper Unicode, quoting  https://en.wikipedia.org/wiki/C0_and_C1_control_codes: "The C1 codes require two bytes to be encoded in UTF-8".

So the code in yayl should be updated to also escape characters above 0x80.

If you can point me in the direction of the yajl library being used, I would gladly supply a patch.

### PR DESCRIPTION
**Pull request by mchro**
Also fix test harness for API. This should fix https://github.com/edenhill/kafkacat/issues/196

### CODE REVIEW

**Comment by alexey-tr** [comment]
@edenhill, is there a chance that this PR will be merged ?

File: test/api/gen-extra-close.c [hunk 1]
```diff
@@ -15,5 +15,5 @@ int main(void) {
   CHK(yajl_gen_map_close(yg));
   s = yajl_gen_map_close(yg);
 
-  return (yajl_gen_generation_complete == s);
+  return !(yajl_gen_generation_complete == s);
```
Comment (edenhill) [inline test/api/gen-extra-close.c:18]: What's this for?

**Review by edenhill — APPROVED** [review]
Great stuff!

File: test/api/gen-extra-close.c (hunk 1 above)
Comment (mchro) [reply test/api/gen-extra-close.c:18]: Trying to remember; I'm pretty sure this was related to the missing check of the exit code - this test could effectively never fail.

**Comment by edenhill** [comment]
thank you :+1:

**Pull request merged**